File: org.kevoree.modeling.microframework/src/test/java/org/kevoree/modeling/microframework/test/cloud/CloudDimension.java
package org.kevoree.modeling.microframework.test.cloud;

import org.kevoree.modeling.api.KUniverse;
import org.kevoree.modeling.api.abs.AbstractKDimension;
import org.kevoree.modeling.microframework.test.cloud.impl.CloudViewImpl;

/**
 * Created by duke on 10/13/14.
 */
public class CloudDimension extends AbstractKDimension<CloudView, CloudDimension, CloudUniverse> {

    protected CloudDimension(KUniverse univers, long key) {
        super(univers, key);
    }

    @Override
    protected CloudView internal_create(Long timePoint) {
        return new CloudViewImpl(timePoint,this);
    }

}

File: org.kevoree.modeling.microframework.typescript/src/test/ts/org/kevoree/modeling/microframework/test/cloud/CloudDimension.ts
///<reference path="../../../api/KUniverse.ts"/>
///<reference path="../../../api/abs/AbstractKDimension.ts"/>
///<reference path="impl/CloudViewImpl.ts"/>

class CloudDimension extends AbstractKDimension<CloudView, CloudDimension, CloudUniverse> {

  constructor(univers: KUniverse, key: number) {
    super(univers, key);
  }

  public internal_create(timePoint: number): CloudView {
    return new CloudViewImpl(timePoint, this);
  }

}

